core/src/main/java/org/elasticsearch/index/mapper/DocumentParser.java

```diff
@@ -676,16 +676,6 @@ private static void parseNullValue(ParseContext context, ObjectMapper parentMapp
 
     private static Mapper.Builder<?,?> createBuilderFromDynamicValue(final ParseContext context, XContentParser.Token token, String currentFieldName) throws IOException {
         if (token == XContentParser.Token.VALUE_STRING) {
-            // do a quick test to see if its fits a dynamic template, if so, use it.
-            // we need to do it here so we can handle things like attachment templates, where calling
-            // text (to see if its a date) causes the binary value to be cleared
-            {
-                Mapper.Builder builder = context.root().findTemplateBuilder(context, currentFieldName, "text", null);
-                if (builder != null) {
-                    return builder;
-                }
-            }
```

Comment:
This code was initially added to fix #1418. But I checked and the bug does not reproduce without this block of code with recent versions of Jackson.